library/cartridge_set_needs_restart.py

```diff
@@ -154,10 +155,48 @@ def needs_restart(params):
         # it mean that instance should be restarted to apply change
         if new_value is not None:
             if current_cfg.get(param_name) != new_value:
-                return helpers.ModuleRes(facts={'needs_restart': True})
+                return True, None
+
+    return False, None
+
+
+def set_needs_restart(params):
+    instance_info = params['instance_info']
+    console_sock = instance_info['console_sock']
+
+    # check if instance was not started yet
+    if not os.path.exists(console_sock):
+        return helpers.ModuleRes(facts={'needs_restart': True})
+
+    try:
+        control_console = helpers.get_control_console(console_sock)
+    except helpers.CartridgeException as e:
+        allowed_errcodes = [
+            helpers.cartridge_errcodes.SOCKET_NOT_FOUND,
+            helpers.cartridge_errcodes.FAILED_TO_CONNECT_TO_SOCKET,
+            helpers.cartridge_errcodes.INSTANCE_IS_NOT_STARTED_YET
+        ]
+        if e.code in allowed_errcodes:
+            return helpers.ModuleRes(facts={'needs_restart': True})
+
+        raise e
+
+    if params['check_package_updated']:
+        needs_restart, err = check_needs_restart_to_update_package(params)
+        if err is not None:
+            return helpers.ModuleRes(failed=True, msg=err)
+        if needs_restart:
+            return helpers.ModuleRes(facts={'needs_restart': True})
+
+    if params['check_config_updated']:
+        needs_restart, err = check_needs_restart_to_update_config(params, control_console)
+        if err is not None:
+            return helpers.ModuleRes(failed=True, msg=err)
+        if needs_restart:
+            return helpers.ModuleRes(facts={'needs_restart': True})
 
-    return helpers.ModuleRes(changed=False)
+    return helpers.ModuleRes(changed=False, facts={'needs_restart': False})
```

Comment:
`changed=False` seems to be redundant here.

Reply:
No, because by default it's false